contributor-docs/cicd.md

```diff
@@ -0,0 +1,96 @@
+# CI/CD
+
+This repository uses [GitHub Actions](https://docs.github.com/en/actions) for all CI/CD as well as for some task automation.
+
+All workflows rely on the [setup-env action](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/.github/actions/setup-env/action.yml)
+to allow them to invoke CI/CD Golang scripts from the [ci-cd directory](https://github.com/GoogleCloudPlatform/DataflowTemplates/tree/main/cicd).
+Any new CI/CD commands should be added to that directory.
+
+## Adding new workflows
+
+All templates tests should be exercised via the [java-pr workflow](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/.github/workflows/java-pr.yml) by default.
+This executes on all pull requests and pushes to main, as well as on a twice-daily schedule.
+Results from this workflow can be found under the [actions tab](https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/workflows/java-pr.yml).
+
+Splitting one or more templates' tests into a separate workflow can help workflows run significantly more quickly since they don't require building all modules.
+To split a set of templates into their own CI workflow, follow these steps:
+
+1) Add an alias to the templates you would like to build/test together in https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/cicd/internal/flags/common-flags.go#L32
+2) Add a new workflow to the `.github/` folder for your tests. This can be a copy of the main [java-pr workflow](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/.github/workflows/java-pr.yml) with a few changes:
+
+- Update the [pull_request trigger paths](https://github.com/manitgupta/DataflowTemplates/blob/3f8efd4f424df5a6b1e3647aa3717eb04348beaf/.github/workflows/java-pr.yml#L23) to point to the templates you want to run your tests against.
+- Update all `./cicd/<script>` calls with a pointer to the alias you defined in (1). For example `./cicd/run-build --modules-to-build="<alias>"`
+
+3) Exclude the paths you added in (2) from the main [java-pr pull_request trigger paths](https://github.com/manitgupta/DataflowTemplates/blob/3f8efd4f424df5a6b1e3647aa3717eb04348beaf/.github/workflows/java-pr.yml#L23)
+
+See https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1573 for an example of splitting out the Spanner tests.
+
+## Runners
+
+There are 2 types of runners that CI/CD runs are performed on: GitHub hosted and self-hosted. These are configured in the `runs-on` section of each workflow. For example, `runs-on: [self-hosted, perf]` means that the workflow is running on a self-hosted runner for perf tests.
+
+For more information on GitHub hosted runners, see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+
+Google self-hosted runners are provisioned using the scripts defined in `https://github.com/manitgupta/DataflowTemplates/tree/main/.github/scripts`.
```

Comment:
it links to a fork manitgupta/DataflowTemplates. Is this intended? Can we add it to the main repo?

Reply:
Yeah, fixed - thanks